sdk/python/tests/test_client.py

```diff
@@ -307,131 +295,131 @@ def test_get_feature_set(self, mocked_client, mocker):
-                )
-            ),
-        )
-
-        expected_dataframe = pd.DataFrame(
-            {
-                "datetime": [datetime.utcnow() for _ in range(3)],
-                "customer": [1001, 1002, 1003],
-                "transaction": [1001, 1002, 1003],
-                "my_project/customer_feature_1:1": [1001, 1002, 1003],
-                "my_project/customer_feature_2:1": [1001, 1002, 1003],
-            }
-        )
-
-        final_results = tempfile.mktemp()
-        to_avro(file_path_or_buffer=final_results, df=expected_dataframe)
-
-        mocker.patch.object(
-            mocked_client._serving_service_stub,
-            "GetBatchFeatures",
-            return_value=GetBatchFeaturesResponse(
-                job=BatchFeaturesJob(
-                    id="123",
-                    type=JobType.JOB_TYPE_DOWNLOAD,
-                    status=JobStatus.JOB_STATUS_DONE,
-                    file_uris=[f"file://{final_results}"],
-                    data_format=DataFormat.DATA_FORMAT_AVRO,
-                )
-            ),
-        )
-
-        mocker.patch.object(
-            mocked_client._serving_service_stub,
-            "GetJob",
-            return_value=GetJobResponse(
-                job=BatchFeaturesJob(
-                    id="123",
-                    type=JobType.JOB_TYPE_DOWNLOAD,
-                    status=JobStatus.JOB_STATUS_DONE,
-                    file_uris=[f"file://{final_results}"],
-                    data_format=DataFormat.DATA_FORMAT_AVRO,
-                )
-            ),
-        )
-
-        mocker.patch.object(
-            mocked_client._serving_service_stub,
-            "GetFeastServingInfo",
-            return_value=GetFeastServingInfoResponse(
-                job_staging_location=f"file://{tempfile.mkdtemp()}/",
-                type=FeastServingType.FEAST_SERVING_TYPE_BATCH,
-            ),
-        )
-
-        mocked_client.set_project("project1")
-        response = mocked_client.get_batch_features(
-            entity_rows=pd.DataFrame(
-                {
-                    "datetime": [
-                        pd.datetime.now(tz=timezone("Asia/Singapore")) for _ in range(3)
-                    ],
-                    "customer": [1001, 1002, 1003],
-                    "transaction": [1001, 1002, 1003],
-                }
-            ),
-            feature_refs=[
-                "my_project/customer_feature_1:1",
-                "my_project/customer_feature_2:1",
-            ],
-        )  # type: Job
-
-        assert response.id == "123" and response.status == JobStatus.JOB_STATUS_DONE
-
-        actual_dataframe = response.to_dataframe()
-
-        assert actual_dataframe[
-            ["my_project/customer_feature_1:1", "my_project/customer_feature_2:1"]
-        ].equals(
-            expected_dataframe[
-                ["my_project/customer_feature_1:1", "my_project/customer_feature_2:1"]
-            ]
-        )
+    # @pytest.mark.parametrize(
```

Comment:
Why is this commented out rather than removed?

Reply:
Because we need this test. The problem is that the test should be fixed instead of removed entirely, but I think its out of scope for this PR.

Reply:
Alternatively we can delete it and depend entirely on e2e, but I think having unit tests for these methods makes sense. Some parts of this Python SDK are not properly tested though, given the high amount of IO